### PR TITLE
Added object glueing.

### PR DESCRIPTION
**Pull request by xxalpha**
Not a player feature.

Let's you glue ~~objects~~ movable atoms together so they all move at once.

I want this for another PR, but PRing separately.

### CODE REVIEW

**Comment by okaw** [comment]
Is it alike parenting, or is it more of a "weld"?

**Comment by xxalpha** [comment]
Welding I guess?

**Comment by okaw** [comment]
Neat. I feel like this is going to make way for some pretty fun and wacky admin events

**Comment by Supermichael777** [comment]
https://www.youtube.com/watch?v=cg4Z00VTP7I

**Comment by Jordie0608** [comment]
Spaceworm you say?

**Comment by tkdrg** [comment]
@xxalpha is this ready

File: code/game/objects/glueing.dm [hunk 1]
```diff
@@ -0,0 +1,47 @@
+//like buckling, but for objects
+/atom/movable/var/list/glued_objects = list()
+/atom/movable/var/atom/movable/glued_to = null
+
+/atom/movable/proc/glue_object(var/atom/movable/AM)
+	if(!AM || !istype(AM))
```
Comment (RemieRichards) [inline code/game/objects/glueing.dm]: remove the ```!AM```, ```istype()``` has it covered

File: code/game/objects/glueing.dm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		return
+	glued_objects |= AM
+	AM.glued_to = src
+
+/atom/movable/proc/unglue_object(var/atom/movable/AM)
+	if(!AM || !istype(AM))
```
Comment (RemieRichards) [inline code/game/objects/glueing.dm]: remove the ```!AM```, ```istype()``` has it covered

**Comment by xxalpha** [comment]
Yeah this is all there is.

**Comment by octareenroon91** [comment]
Am I looking at a Katamari kind of thing where all the atoms occupy the same loc?

**Comment by xxalpha** [comment]
yes

**Comment by octareenroon91** [comment]
Before I looked at the code I 
was imagining objects on different tiles moving at the same time, like a raft or something.

**Comment by KorPhaeron** [comment]
Katamari?

**Comment by RemieRichards** [comment]
@KorPhaeron Katamari Damacy

**Comment by KorPhaeron** [comment]
@RemieRichards 

I meant that as in "is this katamari" because I didn't actually read the comments

I am well aware of the game, it's how I met my girlfriend

**Comment by duncathan** [comment]
@KorPhaeron did you (literally) bond with her by being rolled up by the same katamari?

**Comment by Dorsisdwarf** [comment]
So, what are the differences between this and buckling, like the turreted ATV?

**Comment by xxalpha** [comment]
You can't buckle objects.

**Comment by Iamgoofball** [comment]
Toolgun when?
Thrusters when?
DarkRP when?

**Pull request closed without merging**